src/main/java/net/ripe/rpki/rsyncit/rrdp/RrdpFetcher.java

```diff
@@ -280,69 +283,29 @@ private ProcessPublishElementResult processPublishElements(Element doc, Optional
         return new ProcessPublishElementResult(objects, collisionCount.get());
     }
 
-    private static Instant extractSigningTime(RpkiSignedObject o) {
-        return Instant.ofEpochMilli(o.getSigningTime().getMillis());
-    }
-
+    /**
+     * Try to get some creation timestamp from the object itself. If it's impossible to parse
+     * the object, use the default (based on the last-modified header of the snapshot).
+     *
+     * @param objectUri uri of object
+     * @param decoded content of object
+     * @param lastModified modification time to use as fallback
+     * @return
+     */
     private Instant getTimestampForObject(final String objectUri, final byte[] decoded, Instant lastModified) {
```

Comment:
I would also rename `lastModified` to `defaultTimestamp` or something like this here, it was an oversight from before.

Reply:
👍